Cython/Utility/ObjectHandling.c

```diff
@@ -443,7 +442,29 @@ static CYTHON_INLINE PyObject *__Pyx_GetItemInt_{{type}}_Fast(PyObject *o, Py_ss
     return PySequence_GetItem(o, i);
 #endif
 }
-{{endfor}}
+
+static CYTHON_INLINE PyObject *__Pyx_GetItemInt_List_Fast(PyObject *o, Py_ssize_t i,
+                                                              CYTHON_NCP_UNUSED int wraparound,
+                                                              CYTHON_NCP_UNUSED int boundscheck) {
+#if CYTHON_ASSUME_SAFE_MACROS && CYTHON_ASSUME_SAFE_SIZE && !CYTHON_AVOID_BORROWED_REFS
+    Py_ssize_t wrapped_i = i;
+    if (wraparound & unlikely(i < 0)) {
+        wrapped_i += PyList_GET_SIZE(o);
+    }
+    if ((!boundscheck) || likely(__Pyx_is_valid_index(wrapped_i, PyList_GET_SIZE(o)))) {
+#if !CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS
+        PyObject *r = PyList_GET_ITEM(o, wrapped_i);
+        Py_INCREF(r);
+#else
+        PyObject *r = PyList_GetItemRef(o, wrapped_i);
```

Comment:
This function is new in Py3.13a4 and needs a version guard.
See https://github.com/python/cpython/commit/d0f1307580a69372611d27b04bbf2551dc85a1ef

Reply:
That said, rather than repeating the "use GET_ITEM+INCREF or GetItemRef" decision in various places, it seems worth having an inline function that does the equivalent of "GET_ITEM+INCREF" efficiently in different environments.